shared/TestStandComm/SerialDevice.h

```diff
@@ -3,16 +3,67 @@
 
 #include <stdint.h>
 
+/**
+ * @class SerialDevice
+ * 
+ * @brief Abstract base class representing an interface to serial communication hardware
+ *        on an arbitrary system
+ * 
+ * This is the lowest layer in the protocol stack and is only responsible for sending
+ * bytes. It performs no interpretation of the data being sent.
+ */
 class SerialDevice
 {
     public:
+        /**
+         * @brief Open a connection to the serial hardware
+         * 
+         * @param baud_rate The baud rate for the serial connection
+         * 
+         * @return true if the connection opened successfully, otherwise false
+         */
         virtual bool ser_connect(uint32_t baud_rate) = 0;
```

Comment:
Since all of these are class methods, i would omit `ser_` in names

Reply:
I would like to, but the problem with that is in the `LinuxSerialDevice` I have to call C standard library functions to actually talk to the hardware and they are called `open`, `read`, `write` etc. so it becomes ambiguous which one I'm calling